File: 231204/그릇 쌓기/dishes.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String arrangement = scanner.nextLine();
        int result = calculateHeight(arrangement);
        System.out.println(result);
    }

    static int calculateHeight(String arrangement) {
        int height = 5;
        Stack<Character> stack = new Stack<>();

        for (char ch : arrangement.toCharArray()) {
            if (ch == '(') {
                stack.push('(');
                height += 10;
            } else {
                if (!stack.isEmpty() && stack.peek() == '(') {
                    stack.pop();
                } else {
                    stack.push(')');
                    height += 5;
                }
            }
        }

        return height;
    }
}

File: 231204/그릇 쌓기/dishes.py
def calculate_height(arrangement):
    height = 0
    stack = []

    for char in arrangement:
        if char == '(':
            stack.append('(')
            height += 10
        else:
            if stack and stack[-1] == '(':
                stack.pop()
            else:
                stack.append(')')
                height += 5

    return height

# 입력 받기
arrangement = input()

# 높이 계산 및 출력
result = calculate_height(arrangement)
print(result)